code/game/objects/items/ipc_overloaders.dm

```diff
@@ -0,0 +1,319 @@
+		TRAIT_OVERLOADER_OD_MEDIUM = TRAIT_OVERLOADER_OD_EFFECT
+	)
+
+	var/static/list/step_down_effects = list(
+		TRAIT_OVERLOADER_OD_EFFECT = TRAIT_OVERLOADER_OD_MEDIUM,
+		TRAIT_OVERLOADER_OD_MEDIUM = TRAIT_OVERLOADER_OD_INITIAL
+	)
+
+/obj/item/ipc_overloader/Initialize()
+	. = ..()
+	item_state = icon_state
+
+/obj/item/ipc_overloader/update_icon()
+	if(uses == initial(uses))
+		icon_state = initial(icon_state)
+	else if(uses == initial(uses) / 2)
+		icon_state = "[initial(icon_state)]-1"
+	else
+		icon_state = "[initial(icon_state)]-spent"
+
+/obj/item/ipc_overloader/examine(mob/user)
+	. = ..()
+	if(Adjacent(user))
+		if(uses)
+			to_chat(user, SPAN_NOTICE("It has <b>[uses]</b> uses left."))
+		else
+			to_chat(user, SPAN_WARNING("It's totally spent."))
+
+/obj/item/ipc_overloader/attack_self(mob/user)
+	if(!uses)
+		to_chat(user, SPAN_WARNING("\The [src] is totally spent."))
+		return
+	if(isipc(user))
+		user.visible_message("<b>[user]</b> jabs themselves with \the [src].", SPAN_NOTICE("You jab yourself with \the [src]."))
+		handle_overloader_effect(user)
+		if(effect_time)
+			addtimer(CALLBACK(src, PROC_REF(midway_overloader_effect), user, effects), effect_time)
+		uses--
+		update_icon()
+		return
+	to_chat(user, SPAN_WARNING("\The [src] has no use for you!"))
+
+/obj/item/ipc_overloader/proc/handle_overloader_effect(var/mob/living/carbon/human/target)
+	handle_overdose(target)
+
+/obj/item/ipc_overloader/proc/midway_overloader_effect(var/mob/living/carbon/human/target, var/effect_amount)
+	if(effect_amount)
+		addtimer(CALLBACK(src, PROC_REF(midway_overloader_effect), target, effect_amount - 1), effect_time)
+		return TRUE
+	addtimer(CALLBACK(src, PROC_REF(finish_overloader_effect), target), effect_time)
+	finish_overloader_effect(target)
+	return FALSE
+
+/obj/item/ipc_overloader/proc/finish_overloader_effect(var/mob/living/carbon/human/target)
+	handle_overdose_stepdown(target)
+
+// use traits to step up and down from the overdose effects
+/obj/item/ipc_overloader/proc/handle_overdose(var/mob/living/carbon/human/target)
+	var/added_trait = FALSE
+	for(var/trait in step_up_effects)
+		if(HAS_TRAIT(target, trait))
+			REMOVE_TRAIT(target, trait, TRAIT_SOURCE_OVERLOADER)
+			ADD_TRAIT(target, step_up_effects[trait], TRAIT_SOURCE_OVERLOADER)
+			added_trait = TRUE
+			break
+	if(HAS_TRAIT(target, TRAIT_OVERLOADER_OD_EFFECT))
+		handle_overdose_effect(target)
+	else if(!added_trait)
+		ADD_TRAIT(target, TRAIT_OVERLOADER_OD_INITIAL, TRAIT_SOURCE_OVERLOADER)
+
+/obj/item/ipc_overloader/proc/handle_overdose_effect(var/mob/living/carbon/human/target)
+	switch(rand(1, 8))
+		if(1) // random limb fire damage
+			to_chat(target, SPAN_WARNING("Your limbs rapidly heat up!"))
+			target.take_overall_damage(0, 20)
+		if(2) // random component damage
+			to_chat(target, SPAN_WARNING("Your internals rapidly heat up!"))
+			var/probability = 100
+			for(var/obj/item/organ/internal/internal_component in shuffle(target.internal_organs))
```

Comment:
```suggestion
			for(var/obj/item/organ/internal/internal_component as anything in shuffle(target.internal_organs))
```

This list only contains stuff of this type right? Or does it contain stuff like implants too, I think it doesn't.